src/com/amazon/corretto/crypto/provider/AmazonCorrettoCryptoProvider.java

```diff
@@ -91,8 +93,15 @@ private void buildServiceMap() {
     addService("KeyFactory", "EC", "EvpKeyFactory$EC");
 
     if (shouldRegisterEdDSA) {
-      addService("KeyFactory", "EdDSA", "EvpKeyFactory$EdDSA");
-      addService("KeyFactory", "Ed25519", "EvpKeyFactory$EdDSA");
+      // KeyFactories are used to convert key encodings to Java Key objects. ACCP's KeyFactory for
+      // Ed25519 returns keys of type EvpEdPublicKey and EvpEdPrivateKey that do not implement
+      // EdECKey interface. One should register the KeyFactories from ACCP if they only use the
+      // output of the factories with ACCP's services.
+      // Once ACCP supports multi-release jar, then this option can be removed.
+      if (shouldRegisterEdKeyFactory) {
```

Comment:
instead of making the consumers think about configuration, can we check java version here and only register if > 15?

Reply:
Nah, that's not enough. If a customer has an application, that needs their key factory to produce Sun JCE keys, they can't use our implementation without changing their code.

Reply:
Ideally, we only want to register Ed25519 related services for JDKs 15+. However, we don't have a systematic way to implement a class that depends on some types that are not available in JDK 8. The systematic solution is to use [multi-release jars](https://openjdk.org/jeps/238).

Reply:
I see, so we'd release a multi-release jar that exposes a SunJCE compatible implementation of Ed25519 for JDK15+. And on lower JDKs we'd remove support? Or would we still be able to provide the kind of support we have now?

Reply:
We can provide the same support as of now. The idea behind multi-release jar is that the same Java class can be implemented in multiple files. Each file is an independent implementation and at runtime, depending on the JDK version, the correct one would be loaded.

Reply:
Are we sure that we need multi-release jars? Is there a way to do this with Java reflection? 

I suspect that there is a way to at runtime check if the class `java.security.interfaces.EdECPublicKey` exists (if it does then ACCP is on JDK >= 15), and instantiate a [Reflection Proxy](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/Proxy.html) and [InvocationHandler](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/reflect/InvocationHandler.html) that implements that interface at runtime. It's hacky, but I think doable.

Reply:
makes sense.